compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -782,7 +782,53 @@ object RefChecks {
         report.error(problem(), clazz.srcPos)
       }
 
+      // Checks base types of the current clazz against each other
+      //
+      // In particular, it checks that there are no two base classes with
+      // different type instantiations.
+      //
+      // ported from Scala 2:
+      //   https://github.com/scala/scala/blob/9bb659e62a9239c01aec14c171f8598bb1a576fe/src/compiler/scala/tools/nsc/typechecker/RefChecks.scala#L834-L883
+      def validateBaseTypes(): Unit = {
+        val tpe = clazz.thisType // in Scala 2 this was clazz.tpe
```

Comment:
Here is also a difference from Scala 2, where I use `thisType` and Scala 2 used `.tpe`.